src/sortStudents.ts

```diff
@@ -1,16 +1,70 @@
-
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desk';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+):Student[] {
+  const studentArr: Student[] = [...students];
+
+  function getGrades(arr: number[]): number {
+    return arr.reduce((a: number, b: number) => a + b) / arr.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+      studentArr.sort((a: Student, b: Student) => (order === 'asc'
+        ? a.name.localeCompare(b.name)
+        : b.name.localeCompare(a.name)));
+      break;
+
+    case SortType.Surname:
+      studentArr.sort((a: Student, b: Student) => (order === 'asc'
+        ? a.surname.localeCompare(b.surname)
+        : b.surname.localeCompare(a.surname)));
+      break;
+
+    case SortType.Age:
+      studentArr.sort((a: Student, b: Student) => (order === 'asc'
+        ? a.age - b.age
+        : b.age - a.age));
+      break;
+
+    case SortType.Married:
+      studentArr.sort((a: Student, b: Student) => {
+        if (a.married === b.married) {
+          return 0;
+        }
+
+        if ((a.married && order === 'asc') || b.married) {
+          return 1;
+        }
+
+        return -1;
+      });
+      break;
```

Comment:
```suggestion
    case SortType.Age:
    case SortType.Married:
      studentArr.sort((a: Student, b: Student) => (order === 'asc'
        ? +a[sortBy] - +b[sortBy]
        : +b[sortBy] - +a[sortBy]));
      break;
```
same for age and married
converting true/false to number return 1 or 0

Reply:
I have some errors if I do like you
![image](https://github.com/mate-academy/ts_sort_students/assets/48822777/02367016-7d3c-4e2f-bafe-e533f21b378e)
![image](https://github.com/mate-academy/ts_sort_students/assets/48822777/246c27c7-dc84-4555-bbfe-5899180c5f80)

How I can fix this problem?